service/service.go

```diff
@@ -260,6 +254,17 @@ func New(config Config) (*Service, error) {
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		meta := loggermeta.New()
+
+		customObject, err := key.ToCustomObject(obj)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		meta.KeyVals["cluster"] = key.ClusterID(customObject)
+
+		ctx = loggermeta.NewContext(ctx, meta)
+
```

Comment:
Here the cluster ID is added to the context.

Reply:
I think you do not need this. @kopiczko added something generic to operatorkit so that the self link of the reconciled object is put automatically to the context. The cluster ID is then part of this since the reconciled object's namespace is the cluster ID. See also https://github.com/giantswarm/operatorkit/pull/117.

Reply:
Thanks, I see the self link now which is v nice. So I've removed the custom loggermeta context.